showcase/tests/integration/components/hds/form/masked-input/base-test.js

```diff
@@ -34,14 +34,22 @@ module(
       assert.dom('.hds-form-visibility-toggle .hds-icon-eye').exists();
     });
 
-    test('it should render readable text when `isContentMasked` is false', async function (assert) {
+    test('it should render readable text when `isContentMasked` is false and visibility should change if `isContentMasked` changes', async function (assert) {
```

Comment:
`@isContentMasked` is an argument that sets [the initial state](https://github.com/hashicorp/design-system/pull/2488/files#diff-3ea421e531b1bede671031b5c7347643eff7b62c0318874ad64df46b6ca68da1R35) (as we're only doing it in the constructor), not an argument that would allow external control of this component.

I see two ways forward here: 
 - we can revert the change in this test file and leave the argument to only handle initial rendering
 - we can introduce the [`_isControlled` pattern we use in the DisclosurePrimitive](https://github.com/hashicorp/design-system/blob/main/packages/components/src/components/hds/disclosure-primitive/index.ts#L39) (for Accordion)

whichever approach we take, it may be a good idea to address these on a component basis

Reply:
@alex-ju i think it is kinda being used to control the component externally... at least thats what I remember from the initial thread https://hashicorp.slack.com/archives/C7KTUHNUS/p1728404801116789?thread_ts=1728404760.159249&cid=C7KTUHNUS

I'll make it follow the `_isControlled` pattern..

Reply:
open to other patterns as well, if we find a better way to allow for both internal and external control of certain arguments

Reply:
See comment above, about the fact that the current implementation may not work as expected (and in any case, it should be covered better by testing).